IF/src/main/java/com/github/stefvanschie/inventoryframework/gui/type/ChestGui.java

```diff
@@ -113,9 +113,14 @@ public void update() {
         super.updating = true;
 
         if (isDirty() || dirtyRows) {
+            Inventory oldInventory = this.inventory;
             this.inventory = createInventory();
             this.dirtyRows = false;
 
+            for (HumanEntity viewer : new ArrayList<>(oldInventory.getViewers())) {

```

Comment:
`oldInventory.getViewers()` might cause a `NullPointerException` when the inventory is shown for the first time, when `this.inventory` is `null`. Please add a guard against this.